libsoftwarecontainer/unit-test/networkgatewayparser_unittest.cpp

```diff
@@ -0,0 +1,171 @@
+
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#include "gateway/networkgatewayparser.h"
+#include "softwarecontainer-common.h"
+#include "gateway/iptableentry.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include <jansson.h>
+
+class MockNetworkGatewayParser :
+    public NetworkGatewayParser
+{
+public:
+    MockNetworkGatewayParser() { };
+};
+
+class NetworkGatewayParserTest : public ::testing::Test
+{
+protected:
+    ::testing::NiceMock<MockNetworkGatewayParser> networkParser;
```

Comment:
This mock class doesn't mock anything, so you could use the NetworkGatewayParser directly and remove the Mock class, I think.